scipy/linalg/matfuncs.py

```diff
@@ -668,3 +672,191 @@ def rounded_sign(x):
         return S0
     else:
         return S0, errest
+
+
+def is_matrix_symmetric(A, rtol=1e-05, atol=1e-08):
+    """
+    Returns True if a matrix is symmetric
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    rtol : float
+        Relative tolerance.
+    atol : float
+        Absolute tolerance.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is symmetric.
+
+    Examples
+    --------
+    from scipy.linalg import is_matrix_symmetric
+    m = np.array([[2,3],[3,2]])
+    print(is_matrix_symmetric(m))
+    m = np.array([[5,0],[1,3]])
+    print(is_matrix_symmetric(m))
+    References
+    ----------
+    Hazewinkel, Michiel, ed. (2001) [1994], "Symmetric matrix", Encyclopedia of Mathematics, 
+    Springer Science+Business Media B.V. / Kluwer Academic Publishers, ISBN 978-1-55608-010-4
+    credit to: Nils Werner
+    https://stackoverflow.com/questions/42908334/checking-if-a-matrix-is-symmetric-in-numpy
+    """
+    A = _asarray_square(A)
+    return np.allclose(A, A.T, rtol=rtol, atol=atol)
+
+def is_matrix_positive_definite(A):
+    """
+    Returns True if a matrix is positive definite
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is positive definite.
+
+    Examples
+    --------
+    from scipy.linalg import is_matrix_positive_definite
+    m = np.array([[5,1],[1,3]])
+    print(is_matrix_positive_definite(m))
+    m = np.array([[1, -1, 0],[-1, 5, 0],[0, 0, 7]])
+    print(is_matrix_positive_definite(m))
+
+    References
+    ----------
+    Definiteness of a matrix, From Wikipedia, the free encyclopedia
+    https://en.wikipedia.org/wiki/Positive-definite_matrix#Eigenvalues
+    """
+    A = _asarray_square(A)
+    if not is_matrix_symmetric(A):
+        raise ValueError('expected symmetric matrix')
+    eiVal, eiVecR=eig(A)
+    return (eiVal>0).all()
```

Comment:
Same as above. If the array is too large it can always have some numerical noise around the small eigenvalues. Hence it is tricky for testing posdefness. That's why Cholesky decomposition is used for testing definiteness.

Reply:
Cholesky decomposition for testing definiteness has issues with round off errors that eigenvalues can overcome. Both solutions are far from ideal, however they are simple to implement. 
"In order to perform Cholesky Decomposition of a matrix, the matrix has to be a positive definite matrix." And to test for definiteness, one uses Cholesky Decomposition. It sounds recursive and it does not test for semi positive or for negative definiteness, which can be done with eigenvalues.

Reply:
Semidefiniteness is much harder to test since it looks almost exact zeros.

Reply:
Cholesky is cheaper to check but you are already eating away its advantage by computing eigenvalues which are more expensive.

Reply:
You're right. So, I changed the code accordingly.